### PR TITLE
Build: Bump actions/labeler from 4 to 5

### PR DESCRIPTION
**Pull request by panbingkun**
Bumps [actions/labeler](https://github.com/actions/labeler) from 4 to 5.

Release notes
https://github.com/actions/labeler/releases/tag/v5.0.0
image

### CODE REVIEW

**Comment by panbingkun** [comment]
The above modifications are based on the following document:
https://github.com/actions/labeler/tree/main?tab=readme-ov-file#usage

Some features have been validated in private repo
https://github.com/panbingkun/github-action-test/pull/2
https://github.com/panbingkun/github-action-test/pull/3

**Review by ajantha-bhat — APPROVED** [review]

File: .github/labeler.yml [hunk 1]
```diff
@@ -17,71 +17,171 @@
...
+    - any-glob-to-any-file: [
+      '.asf.yaml',
+      '.gitattributes',
+      '.gitignore',
+      'baseline.gradle',
+      'deploy.gradle',
+      'jitpack.yml',
+      'travis.yml',
+      '.baseline/**/*',
+      '.github/**/*'
```
Comment (ajantha-bhat) [inline .github/labeler.yml:32]:
```suggestion
      '.github/**/*',
```

**Comment by panbingkun** [comment]
Prove from https://github.com/apache/spark/pull/44350 and https://github.com/apache/kyuubi/pull/5874 that this change should be correct and effective.
Both of the above two are merged and running well.

**Comment by panbingkun** [comment]
Friendly ping @nastra, can you take some time to help review this PR? Thanks!

**Review by nastra — APPROVED** [review]

**Pull request merged**